lib/nxt_http_client/config.rb

```diff
@@ -12,4 +13,18 @@ def dup
       self.class.new(**to_h.deep_dup)
     end
   end
+
+  def build_thread_cache_key
+    "#{SecureRandom.base58}::#{DateTime.current}"
+  end
+
+  def set_thread_cache_key
+    Thread.current[THREAD_CACHE_KEY] ||= build_thread_cache_key
+  end
+
+  def clear_thread_cache_key
+    Thread.current[THREAD_CACHE_KEY] = nil
```

Comment:
What do you think about using [RequestStore gem](https://github.com/steveklabnik/request_store)?